public/components/CampaignAnalytics/Analytics/CampaignReferrals.js

```diff
@@ -5,60 +5,77 @@ import {getCampaignReferrals, setToggleNode} from "../../../actions/CampaignActi
 import CampaignReferralHeader from "./CampaignReferralHeader";
 import InfinityMenu from "react-infinity-menu";
 import ProgressSpinner from "../../utils/ProgressSpinner";
+import DateRangeEditor from "./DateRangeEditor";
 
 class CampaignReferrals extends React.Component {
 
+  constructor(props) {
+    super(props);
+    this.state = {
+      dateRange: {
+        startDate: DateRangeEditor.toDate(this.props.campaign.startDate),
+        endDate: DateRangeEditor.toDate(this.props.campaign.endDate)
+      }
+    };
+  }
+
   componentWillMount() {
-    this.props.campaignReferralActions.getCampaignReferrals(this.props.campaign.id);
+    this.props.campaignReferralActions.getCampaignReferrals(this.props.campaign.id, this.state.dateRange);
   }
 
-  componentWillReceiveProps(nextProps) {
-    if (nextProps.campaign.id !== this.props.campaign.id) {
-      this.props.campaignReferralActions.getCampaignReferrals(nextProps.campaign.id);
+  componentWillUpdate(nextProps, nextState) {
+    if (nextState.dateRange !== this.state.dateRange) {
```

Comment:
what happened to the campaign id test?

Reply:
I don't think we need it, but will reinstate if anything strange happens.